### ISSUE TITLE
title / lastMsg empty

### ISSUE THREAD

**Issue by bmbraga** [bruno222/whatsapp-web-bot#112]
Can someone help me with this problem below?

It includes a message to be answered, but everything is apparently ignored, the return is always blank.

Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
VM65:186 Thu May 14 2020 09:21:50 GMT-0300 (Horário Padrão de Brasília) "nothing to do now... (1)" 19 undefined

**Comment by Logicbloke** [comment]
What commands are you sending?

**Comment by bmbraga** [comment]
if (lastMsg.toLowerCase().indexOf('Eu Quero') > -1)

It always returns blank, as not reading any messages received on the console.

**Comment by Logicbloke** [comment]
What line number is this code from?

**Comment by bmbraga** [comment]
line 231

**Comment by bmbraga** [comment]
I did a test now, in a browser with normal whatsapp it works, in the other with whatsapp business it doesn't work.

**Comment by bmbraga** [comment]
I did a test now, in a browser with normal whatsapp it works, in the other with whatsapp business it doesn't work.

**Comment by bmbraga** [comment]
There must be some different tag in the web interface.

**Comment by neo3** [comment]
WhatsApp Business there is one more <div>, I use this mapping to work:

```
var elementConfig = {
    'chats': [0, 0, 5, 2, 0, 3, 0, 0, 0],
    'chat_icons': [0, 0, 0, 1, 1, 1, 0],
    'chat_title': [0, 0, 0, 1, 0, 0, 0, 0],
    'chat_lastmsg': [0, 0, 0, 1, 1, 0, 0],
    'chat_active': [0, 0],
    'selected_title': [0, 0, 5, 3, 0, 1, 1, 0, 0, 0, 0]
};
```

Does anyone know how I could do to upload an image?

**Comment by bmbraga** [comment]
> WhatsApp Business there is one more **div**, I use this mapping to work:
> 
> ```
> var elementConfig = {
>     'chats': [0, 0, 5, 2, 0, 3, 0, 0, 0],
>     'chat_icons': [0, 0, 0, 1, 1, 1, 0],
>     'chat_title': [0, 0, 0, 1, 0, 0, 0, 0],
>     'chat_lastmsg': [0, 0, 0, 1, 1, 0, 0],
>     'chat_active': [0, 0],
>     'selected_title': [0, 0, 5, 3, 0, 1, 1, 0, 0, 0, 0]
> };
> ```

Thanks, it worked.

**Comment by bmbraga** [comment]
There is still something wrong, because you are sending a reply to messages already read from the body of the chat, and not just the last one, would it be the map? WhatsApp Business

**Comment by neo3** [comment]
Here the Bot get only unread messages.

**Comment by bmbraga** [comment]
> Here the Bot get only unread messages.

Can you send your current code to merge? or, I send you my code.

**Comment by Logicbloke** [comment]
Is the mapping different from the regular Whatsapp? Should we add a condition to test if it's Whatsapp Business to apply the different mapping?

**Comment by 7ayadi** [comment]
> Can someone help me with this problem below?
> 
> It includes a message to be answered, but everything is apparently ignored, the return is always blank.
> 
> Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
> VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
> VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
> VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
> VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
> VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
> VM65:232 Thu May 14 2020 09:21:49 GMT-0300 (Horário Padrão de Brasília) "new message ignored -> " "" ""
> VM65:186 Thu May 14 2020 09:21:50 GMT-0300 (Horário Padrão de Brasília) "nothing to do now... (1)" 19 undefined



> > Here the Bot get only unread messages.
> 
> Can you send your current code to merge? or, I send you my code.

how to connection in mysql ?

**Comment by 7ayadi** [comment]
how to connection in database mysql

**Comment by helders** [comment]
> how to connection in database mysql

@7ayadi you cannot connect directly to a MySQL database, but you could create a backend in any languange you want to exposes API endpoints and use Javascript to send and retrieve information to it using AJAX calls. But since you'll  be making request to another domain you'll have to disable CORS restriction on the browser. You won't be able to do it on Chrome but using Firefox you can do that.

**Comment by andersonkida** [comment]
@helders você chegou a fazer um backend ??

**Comment by helders** [comment]
@andersonkida sim, utilizei uma aplicação que já possuía com o script consumindo endpoints via AJAX, fazendo uma maquina de estado e interagindo da forma que eu necessitava.

yes, i used an application i already had with this script consuming it via AJAX, creating a state machine and interacting the way i wanted it to

**Comment by andersonkida** [comment]
> @andersonkida sim, utilizei uma aplicação que já possuía com o script consumindo endpoints via AJAX, fazendo uma maquina de estado e interagindo da forma que eu necessitava.
> 
> yes, i used an application i already had with this script consuming it via AJAX, creating a state machine and interacting the way i wanted it to

Entendi. Você pode disponibilizar esse backend?

**Comment by helders** [comment]
@andersonkida infelizmente não, é uma aplicação proprietária.
